Source/WebAssembly/WebAssemblyTextPrint.cpp

```diff
@@ -11,6 +11,22 @@
 
 using namespace AST;
 
+bool startsWith(const char *str, const char *prefix)
+{
+        size_t lenstr = strlen(str);
+        size_t lenprefix = strlen(prefix);
+        if(lenstr < lenprefix) { return false; }
+	return (strncmp(prefix, str, lenprefix) == 0);
+}
+
+bool nameConflict(const char *name)
+{
+	return (startsWith(name, "local") ||
+		startsWith(name, "func") ||
+		startsWith(name, "sig") ||
+		startsWith(name, "label"));
```

Comment:
On second thought, what I suggested doesn't work. Consider this case:
```
(func $_func0 ...)
(func $func0 ...)
```

If you import and reexport this module it will export a module with both functions named the same thing:
```
(func $_func0 ...)
(func $_func0 ...)
```

Maybe it should just emit indices instead of names where the AST doesn't have a name.